src/worker/upstreamcommands.js

```diff
@@ -36,41 +51,101 @@ commands['CAP'] = async function(msg, con) {
             await con.state.tempSet('caps_receiving', null);
         }
 
-        let wantedCaps = [
-            'server-time',
-            'multi-prefix',
-            'away-notify',
-            'account-notify',
-            'account-tag',
-            'extended-join',
-            'userhost-in-names',
-            'sasl',
-        ];
+        let requestingCaps = offeredCaps.filter((cap) => wantedCaps.includes(cap.split('=')[0].toLowerCase()))
+                                        .map((cap) => cap.split('=')[0]);
+        let forwardToClient = []
 
         await hooks.emit('cap_to_upstream', {
             client: con,
             message: msg,
-            requesting: wantedCaps,
+            requesting: requestingCaps,
             offered: offeredCaps,
+            forwardToClient: [],
```

Comment:
Just in the middle of tidying some things up, but what is `forwardToClient` used for? It seems to be defined, checked against at line 72 but then never actually used. The same logic used further down at line 100

Reply:
(ps. I'm in the middle of a refactor so don't push anything atm pls)

Reply:
It's used down at line 74, basically it's so that once we get the `CAP LS` from upstream, hooks can advertise capabilities to end users. When `CAP DEL` comes through, the hooks can also 'del' capabilities to clients and when `CAP NEW` comes through, hooks can 'new' capabilities. I wasn't sure whether it'd be best to allow three separate lists on each hook, so just went with this.